socotratools/socotratools/client.py

```diff
@@ -256,3 +259,25 @@ def price_policy(self, locator):
 
     def uw_policy(self, locator):
         return self.__get("/policy/" + locator + "/automatedUnderwritingResult")
+
+    def preview_endorsement_price(self, policy_locator, endorsement_name,
```

Comment:
Can you group these update fields in the parameters and dictionary below by object type? e.g first "fieldValues", then add/update/remove field groups, then add/end exposures